addon/corpus.js

```diff
@@ -1,106 +1,231 @@
-async function freezeAllPages() {
-    // Grey out Freeze button:
-    document.getElementById('freeze').disabled = true;
+let gUrls;  // Array of [filename, url] to freeze.
+let gUrlIndex;  // Pointer to current url in gUrls.
+let gFreezeOptions; // Freeze-dry options from UI.
+let gTimeout; // Load+Freeze timeout from UI.
+
+async function freezeAllPages(event) {
+    event.preventDefault();
+
+    // Check form validity.
```

Comment:
Rather than having a user click the button and then nothing happen (with no further explanation), we could disable the button in an onChange handler of the fields if they're not valid. Then the user gets an additional little prod right after he makes the mistake.

Reply:
the validation errors will be displayed by the browser; we don't need to do anything else here.